app/assets/javascripts/govuk_publishing_components/components/step-by-step-nav.js

```diff
@@ -78,6 +79,12 @@ window.GOVUK.Modules = window.GOVUK.Modules || {};
 
     this.$module.insertBefore(showAll, steps)
     this.$module.showOrHideAllButton = this.$module.querySelectorAll('.js-step-controls-button')[0]
+
+    if (showAllGa4Data) {
+      this.$module.showOrHideAllButton.setAttribute('data-ga4', showAllGa4Data)
+      this.$module.showOrHideAllButton.setAttribute('data-ga4-expandable', '') // Enables tracking of aria-expanded in ga4-event-tracker.
+      this.$module.removeAttribute('data-ga4-show-all-attributes') // Remove it to prevent repetition. The JSON lives there temporarily until the HTML it is needed on is rendered by this JS.
+    }
```

Comment:
As per a discussion between @AshGDS and myself, a test will be added to make sure that the ga4 data attributes are being removed and applied correctly